### PR TITLE
try atom mapping for missing atom in source

### PR DESCRIPTION
**Pull request by jbwheatley**
#8 attempts to use transmutations in scope when atoms/molecules in target are not present in source during mixing.

### CODE REVIEW

File: alphabet-soup/src/main/scala/io/typechecked/alphabetsoup/SelectOrTransmuteOrDefault.scala [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package io.typechecked
+package alphabetsoup
+
+/**
+ * Tries to use SelectFromAtomised to produce a value U from L. If it fails, tries to find a default atom
```
Comment (mcgizzle) [inline alphabet-soup/src/main/scala/io/typechecked/alphabetsoup/SelectOrTransmuteOrDefault.scala]: Docstring should be updated

**Review by mcgizzle — CHANGES_REQUESTED** [review]

File: alphabet-soup/src/test/scala/io/typechecked/alphabetsoup/SelectOrTransmuteOrDefaultSpec.scala [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    implicit val transmute = Transmute[String, Int](_.length)
+    type L = String :: String :: HNil
+    type U = Int
+
+    val l = "hi" :: "twine" :: HNil
+
+    SelectOrTransmuteOrDefault[L, U].apply(l) shouldBe 2
+  }
+
+  it should "try to perform transmutations before inserting defaults" in {
```
Comment (jdrphillips) [inline alphabet-soup/src/test/scala/io/typechecked/alphabetsoup/SelectOrTransmuteOrDefaultSpec.scala]: As discussed transmute and default cases should be swapped around

File: alphabet-soup/src/test/scala/io/typechecked/alphabetsoup/SelectOrTransmuteOrDefaultSpec.scala [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    implicit val transmute = Transmute[String, Boolean](_ => true)
+    type L = Int :: (Int :: String :: HNil) :: HNil
+    type U = Boolean
+
+    val l = 17 :: (14 :: "twine" :: HNil) :: HNil
+
+    SelectOrTransmuteOrDefault[L, U].apply(l) shouldBe true
+  }
+
+  it should "for type not present select and map molecule functorally if there is a transmutation in scope" in {
```
Comment (jdrphillips) [inline alphabet-soup/src/test/scala/io/typechecked/alphabetsoup/SelectOrTransmuteOrDefaultSpec.scala]: It's not mapping the molecule functorally in this example

Comment (jdrphillips) [reply alphabet-soup/src/test/scala/io/typechecked/alphabetsoup/SelectOrTransmuteOrDefaultSpec.scala]: Yes it is! Nevermind

File: alphabet-soup/src/test/scala/io/typechecked/alphabetsoup/SelectOrTransmuteOrDefaultSpec.scala [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    implicit val bob: Transmute[String, Boolean] = Transmute.apply[String, Boolean](_ => true)
+    type L = Int :: String :: HNil
+    type U = Boolean
+
+    val l = 17 :: "twine" :: HNil
+
+    SelectOrTransmuteOrDefault[L, U].apply(l) shouldBe true
+  }
+
+}
```
Comment (jdrphillips) [inline alphabet-soup/src/test/scala/io/typechecked/alphabetsoup/SelectOrTransmuteOrDefaultSpec.scala]: Could you add a case proving Transmutations are not transitive?

`Transumte[A, B]` and `Transmute[B, C]` does not allow you `Transmute[A, C]`

File: alphabet-soup/src/main/scala/io/typechecked/alphabetsoup/SelectOrTransmuteOrDefault.scala [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+ *
+ * This assumes L is atomised.
+ *
+ * ### Replace behaviour
+ *
+ * If we can select U from L, replace acts as it does for SelectFromAtomised
+ *
+ * If we can not select U from L and rely on an implicit default, we replace nothing and return our L unchanged
+ */
+trait SelectOrTransmuteOrDefault[L, U] {
```
Comment (jdrphillips) [inline alphabet-soup/src/main/scala/io/typechecked/alphabetsoup/SelectOrTransmuteOrDefault.scala]: Renamed to match the new order of precedence

File: alphabet-soup/src/main/scala/io/typechecked/alphabetsoup/SelectAndTransmute.scala [hunk 6]
```diff
@@ -0,0 +1,37 @@
+package io.typechecked.alphabetsoup
+
+import shapeless.{::, HList}
+
+trait SelectAndTransmute[L, U] {
+  def apply(l: L): U
+}
+
+object SelectAndTransmute extends LowPrioritySelectAndTransmute {
```
Comment (jdrphillips) [inline alphabet-soup/src/main/scala/io/typechecked/alphabetsoup/SelectAndTransmute.scala]: Could this be named `SelectTransmuted`?

File: alphabet-soup/src/main/scala/io/typechecked/alphabetsoup/Mixer.scala [hunk 7]
```diff
@@ -136,9 +136,9 @@ object MixerImplFromAtomised extends LowPriorityMFAImplicits1 {
   }
 
   implicit def bHeadIsAtomRecurse[A, BH, BT <: HList](
-    implicit atom: Atom[BH],
-    s: SelectOrDefault[A, BH],
-    m2: MixerImplFromAtomised[A, BT]
+                                                       implicit atom: Atom[BH],
```
Comment (jdrphillips) [inline alphabet-soup/src/main/scala/io/typechecked/alphabetsoup/Mixer.scala]: Jack 😭

**Review by jdrphillips — APPROVED** [review]

**Review by mcgizzle — APPROVED** [review]

**Pull request merged**